sdk/iot/hub/src/az_iot_hub_client.c

```diff
@@ -101,6 +104,81 @@ AZ_NODISCARD az_result az_iot_hub_client_get_user_name(
   return AZ_OK;
 }
 
+// TODO: GH #812 (Revisit this temporary function for PnP GA)
+AZ_NODISCARD az_result az_iot_hub_client_get_user_name_with_model_id(
+    az_iot_hub_client const* client,
+    az_span model_id,
+    char* mqtt_user_name,
+    size_t mqtt_user_name_size,
+    size_t* out_mqtt_user_name_length)
+{
+  _az_PRECONDITION_NOT_NULL(client);
+  _az_PRECONDITION_NOT_NULL(mqtt_user_name);
+  _az_PRECONDITION(mqtt_user_name_size > 0);
+
+  const az_span* const module_id = &(client->_internal.options.module_id);
+  const az_span* const user_agent = &(client->_internal.options.user_agent);
+
+  az_span mqtt_user_name_span
+      = az_span_init((uint8_t*)mqtt_user_name, (int32_t)mqtt_user_name_size);
+
+  int32_t required_length = az_span_size(client->_internal.iot_hub_hostname)
+      + (int32_t)sizeof(hub_client_forward_slash) + az_span_size(client->_internal.device_id)
+      + az_span_size(hub_service_preview_api_version)
+      + az_span_size(hub_client_param_separator_span) + az_span_size(hub_digital_twin_model_id)
+      + az_span_size(hub_client_param_equals_span);
```

Comment:
model_id len needs to in here too?

Reply:
No its seems that way. But it is URL encoded later on line 163:
https://github.com/Azure/azure-sdk-for-c/blob/ba1bf8466a774b90c677ff27ef81ea7dd1f7a45b/sdk/iot/hub/src/az_iot_hub_client.c#L163
We don't know the size of the url encoded version so we have to do the validation there.